src/Sessions.h

```diff
@@ -93,8 +93,13 @@ class NetSessions {
 	 * Main entry point for processing packets destined for session analyzers. This
 	 * method is called by the packet analysis manager when after it has processed
 	 * an IP-based packet, and shouldn't be called directly from other places.
+	 *
+	 * @param t The timestamp for this packet.
+	 * @param pkt The packet being processed.
+	 * @param len The number of bytes that haven't been processed yet by packet
+	 * analysis.
 	 */
-	void DoNextPacket(double t, const Packet *pkt);
+	void DoNextPacket(double t, const Packet *pkt, size_t len);
```

Comment:
If I'm not missing anything, we now have exactly one place only that is calling `DoNextPacket`, and that's the IP analyzer forwarding TCP/UDP/ICMP. With that, it seems we can skip some of the re-processing of IP at the beginning of that method by just passing in the protocol directly. How about we change this method to `ProcessTransportLayer(double t, size_t len, const uint8_t* data, Packet* packet, int proto)` and then start directly with the transport-layer logic.

Reply:
I think it's a little weird to pass the packet object but then also pass a bunch of fields that are contained in the packet object. This code is all going to get reworked with the TCP/UDP work after the 4.0 release, so I'm a little inclined to just leave it. At the very most, I can see removing that header length check from DoNextPacket, but that's it.

Reply:
Want to rename the method at least to better express what it's doing?